pkg/provider/passworddepot/passworddepot_api.go

```diff
@@ -0,0 +1,316 @@
+	}
+	return api, nil
+}
+
+func (api *API) doAuthenticatedRequest(r *http.Request) (*http.Response, error) {
+	r.Header.Add("access_token", api.secret.AccessToken)
+	r.Header.Add("client_id", api.secret.ClientID)
+
+	return api.client.Do(r)
+}
+func (api *API) getDatabaseFingerprint(database string) (string, error) {
+	databases, err := api.ListDatabases()
+	if err != nil {
+		return "", fmt.Errorf("error: getting database list: %w", err)
+	}
+
+	for _, db := range databases.Databases {
+		if strings.Contains(db.Name, database) {
+			return db.Fingerprint, nil
+		}
+	}
+
+	return "", errDBNotFound
+}
+
+func (api *API) getSecretFingerprint(databaseFingerprint, secretName, folder string) (string, error) {
+	secrets, err := api.ListSecrets(databaseFingerprint, folder)
+	if err != nil {
+		return "", fmt.Errorf("error: getting secrets list: %w", err)
+	}
+
+	parts := strings.Split(secretName, ".")
+	searchName := parts[0]
+	var fingerprint string
+	for _, entry := range secrets.Entries {
+		if strings.Contains(entry.Name, searchName) {
+			fingerprint = entry.Fingerprint
+			if len(parts) > 1 {
+				return api.getSecretFingerprint(databaseFingerprint, strings.Join(parts[1:], "."), fingerprint)
+			}
+			return fingerprint, nil
+		}
+	}
+	return "", errSecretNotFound
+}
+
+func (api *API) getendpointURL(endpoint string) string {
+	return fmt.Sprintf("https://%s:%s/v1.0/%s", api.baseURL, api.hostPort, endpoint)
+}
+
+func (api *API) login(ctx context.Context) error {
+	loginRequest, err := http.NewRequestWithContext(ctx, "GET", api.getendpointURL("login"), http.NoBody)
+	if err != nil {
+		return fmt.Errorf("error creating request: %w", err)
+	}
+	loginRequest.Header.Add("user", api.username)
+	loginRequest.Header.Add("pass", api.password)
+
+	resp, err := api.client.Do(loginRequest) //nolint:bodyclose // linters bug
+	if err != nil {
+		return fmt.Errorf(DoRequestError, err)
+	}
+
+	accessData := AccessData{}
+	err = ReadAndUnmarshal(resp.Body, resp, &accessData)
+	if err != nil {
+		return fmt.Errorf("error: failed to unmarshal response body: %w", err)
+	}
+
+	api.secret = &accessData
+
+	return nil
+}
+
+func (api *API) ListSecrets(dbFingerprint, folder string) (DatabaseEntries, error) {
+	endpointURL := api.getendpointURL(fmt.Sprintf("list?db=%s", dbFingerprint))
+	if folder != "" {
+		endpointURL = fmt.Sprintf("%s&folder=%s", endpointURL, folder)
+	}
+	listSecrets, err := http.NewRequest("GET", endpointURL, http.NoBody)
+	if err != nil {
+		return DatabaseEntries{}, fmt.Errorf("error: creating secrets request: %w", err)
+	}
+
+	respSecretsList, err := api.doAuthenticatedRequest(listSecrets) //nolint:bodyclose // linters bug
+	if err != nil {
+		return DatabaseEntries{}, fmt.Errorf(DoRequestError, err)
+	}
+
+	dbEntries := DatabaseEntries{}
+	err = ReadAndUnmarshal(respSecretsList.Body, respSecretsList, &dbEntries)
+	return dbEntries, err
+}
+
+func ReadAndUnmarshal(content io.ReadCloser, resp *http.Response, target any) error {
```

Comment:
You already passed in the response. :) Just use the response. :D No need to pass in response AND the body separately. :)

Reply:
Ah you're right :D Thank you I kinda just missed that somehow :face_with_spiral_eyes: